### PR TITLE
[SQL Migration] Add support for assessing XEvent session files

### PR DESCRIPTION
**Pull request by raymondtruong**
This PR adds support for assessing extended event session files with a simple collapsable experience. The user can capture ad-hoc queries, etc. using extended events, and any unsupported features will be identified as assessment issues. 

In terms of backend changes, none are needed as this is an extension of the existing NuGet assessment API, simply with a new folder path parameter being passed in. This functionality was added separately a while back.

This is a feature ported from DMA. Existing DMA experience, for reference:

image

New database selection page:

image

Sample assessment result, from an extended events session containing queries which use FASTFIRSTROW:

image

There are some requirements in place when running the capture which need to be documented. Currently links to DMA docs are being used, but we should look into newer, more detailed docs. An aka.ms link is being used so that we can update the documentation without a code change.

### CODE REVIEW

File: extensions/sql-migration/src/constants/strings.ts [hunk 1]
```diff
@@ -76,6 +76,14 @@ export const RUN_VALIDATION = localize('sql.migration.run.validation', "Run vali
 export const DATABASE_FOR_ASSESSMENT_PAGE_TITLE = localize('sql.migration.database.assessment.title', "Databases for assessment");
 export const DATABASE_FOR_ASSESSMENT_DESCRIPTION = localize('sql.migration.database.assessment.description', "Select the databases that you want to assess for migration to Azure SQL.");
 
+// XEvents assessment
+export const SELECT = localize('sql.migration.select', "Select");
+export const XEVENTS_ASSESSMENT_TITLE = localize('sql.migration.database.assessment.xevents.title', "Assess extended event sessions");
+export const XEVENTS_ASSESSMENT_DESCRIPTION = localize('sql.migration.database.assessment.xevents.description', "For the selected databases, you can additionally assess ad-hoc or dynamic SQL queries or any DML statements initiated through the application data layer by providing extended event session files.");
+export const XEVENTS_ASSESSMENT_DESCRIPTION2 = localize('sql.migration.database.assessment.xevents.description2', "If the extended events include features which are unsupported in Azure SQL, the detected issues will appear in the assessments results on the next step of the wizard as impacted objects.");
```
Comment (raymondtruong) [inline extensions/sql-migration/src/constants/strings.ts:83]: todo: typo

File: extensions/sql-migration/src/dialog/assessment/xEventsAssessmentDialog.ts [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+			value: constants.XEVENTS_ASSESSMENT_DESCRIPTION2,
+			CSSStyles: {
+				...styles.BODY_CSS,
+				'margin': '8px 0px 8px 0px',
+			}
+		}).component();
+
+		const moreInfo = _view.modelBuilder.hyperlink().withProps({
+			label: constants.XEVENTS_ASSESSMENT_HELPLINK,
+			url: 'https://learn.microsoft.com/sql/dma/dma-assesssqlonprem?view=sql-server-ver16#add-databases-and-extended-events-trace-to-assess',
```
Comment (raymondtruong) [inline extensions/sql-migration/src/dialog/assessment/xEventsAssessmentDialog.ts:70]: todo: maybe replace with an aka.ms link so we can replace it with updated documentation once it's available without needing to change the code

**Comment by AkshayMata** [comment]
image
this seems a bit difficult from discoverability side of things. would a button be better?

File: extensions/sql-migration/src/dialog/assessment/xEventsAssessmentDialog.ts (hunk 2 above)
Comment (AkshayMata) [reply extensions/sql-migration/src/dialog/assessment/xEventsAssessmentDialog.ts:70]: great idea!

**Comment by raymondtruong** [comment]
> image this seems a bit difficult from discoverability side of things. would a button be better?

yeah, discussing offline right now to see what makes the most sense - custom button on the right side of the navigation bar is an alternative

**Comment by croblesm** [comment]
Hi @raymondtruong! I only have one comment about the link placement from the first screenshot. 
It will be easier for users to discover this new functionality if we place a button in the right-hand side area of the navigation bar.

I just created a mockup example:
image.

**Comment by mayankshekhar-ms** [comment]
We can give a little more context upfront. Sharing an exploration below for how to show this:

image
We don't need to use an additional blade. If the user doesn't want to use it they can collapse it as shown in the screenshot below:
image

**Comment by raymondtruong** [comment]
@AkshayMata @croblesm @mayankshekhar-ms thanks for the feedback, updated to remove the dialog and just use a collapsible control on the database selection page instead. See the updated screenshot in the PR description. I'm running into a weird issue where it's always loading fully expanded (instead of collapsed, which would be ideal for an optional feature) no matter how I configure it, so will reach out to the ADS team for help on this one.

File: extensions/sql-migration/src/wizard/databaseSelectorPage.ts [hunk 3]
```diff
@@ -212,9 +222,75 @@ export class DatabaseSelectorPage extends MigrationWizardPage {
...
+			.withItems([
+				this._xEventsFolderPickerInput,
+				xEventsFolderPickerButton,
+				xEventsFolderPickerClearButton])
+			.component();
+
+		this._xEventsGroup = this._view.modelBuilder.groupContainer().withLayout({
+			header: constants.XEVENTS_ASSESSMENT_TITLE,
+			collapsible: true
+		}).withProps({ collapsed: false, width: 650, }).withItems([xEventsDescription, xEventsInstructions, xEventsFolderPickerContainer]).component();
```
Comment (alanrenmsft) [inline extensions/sql-migration/src/wizard/databaseSelectorPage.ts:289]: when initializing, please only set the collapsed property in the withLayout method.

Comment (raymondtruong) [reply extensions/sql-migration/src/wizard/databaseSelectorPage.ts]: thanks @alanrenmsft, this worked

File: extensions/sql-migration/src/wizard/databaseSelectorPage.ts [hunk 4]
```diff
@@ -212,18 +224,91 @@ export class DatabaseSelectorPage extends MigrationWizardPage {
 		// load unfiltered table list and pre-select list of databases saved in state
 		await this._filterTableList('', this.migrationStateModel._databasesForAssessment);
 
+		const xEventsDescription = this._view.modelBuilder.text()
+			.withProps({
+				value: constants.XEVENTS_ASSESSMENT_DESCRIPTION,
+				width: this.TABLE_WIDTH,
+				CSSStyles: { ...styles.BODY_CSS },
+				links: [{ text: constants.XEVENTS_ASSESSMENT_HELPLINK, url: 'aka.ms/sql-migration-xe-assess' }]
```
Comment (brian-harris) [inline extensions/sql-migration/src/wizard/databaseSelectorPage.ts:232]: is this url missing protocol?
https://aka.ma/sql-migration-xe-assess

**Review by brian-harris — APPROVED** [review]

File: extensions/sql-migration/src/wizard/databaseSelectorPage.ts (hunk 4 above)
Comment (raymondtruong) [reply extensions/sql-migration/src/wizard/databaseSelectorPage.ts:232]: good catch, fixed!

**Review by siyao-Siyang — APPROVED** [review]
Looks good to me

**Pull request merged**